gh-134786: Py_TPFLAGS_MANAGED_WEAKREF and Py_TPFLAGS_MANAGED_DICT must be used only if Py_TPFLAGS_HAVE_GC set (#135863)
python/cpython: Doc/whatsnew/3.15.rst

  (Contributed by Victor Stinner in :gh:`129813`.)

> Limited C API changes

Reviewer: `Py_TPFLAGS_MANAGED_DICT` and `Py_TPFLAGS_MANAGED_WEAKREF` are not limited C API, right?

Author: Yeah, it seems that you are right. I thought that if `PyType_GetFlags` returns flags then all flags are under limited C API. But I checked now and `Py_TPFLAGS_MANAGED_DICT` and `Py_TPFLAGS_MANAGED_WEAKREF` are excluded (they are under `#ifndef Py_LIMITED_API`).

Thanks!

---------------------

> * If the :c:macro:`Py_TPFLAGS_MANAGED_DICT` and :c:macro:`Py_TPFLAGS_MANAGED_WEAKREF`
>   flags are set then :c:macro:`Py_TPFLAGS_HAVE_GC` must be set too.

Reviewer 2: ```suggestion
* If the :c:macro:`Py_TPFLAGS_MANAGED_DICT` or :c:macro:`Py_TPFLAGS_MANAGED_WEAKREF`
  flag is set then :c:macro:`Py_TPFLAGS_HAVE_GC` must be set too.
```

Maybe it should be "or", not "and"?

Author: Thanks, fixed!
